File: 02-java-spring-boot/src/main/java/com/tqi/checkout/domain/PagamentoInternacional.java
package com.tqi.checkout.domain;

// [🟩O] - OPEN-CLOSED PRINCIPLE: A interface 'Estornavel' é estendida para suportar transações 
// internacionais sem que o contrato original de estorno precise ser modificado ou poluído.
public interface PagamentoInternacional extends Estornavel {
    public void processarTransacaoInternacional(String moedaEstrangeira);
}

File: 04-kotlin-spring-boot/src/main/kotlin/com/dprev/checkout/domain/PagamentoInternacional.kt
package com.dprev.checkout.domain

// [🟩O] - OPEN-CLOSED PRINCIPLE: A interface 'Estornavel' é estendida para suportar transações 
// internacionais sem que o contrato original de estorno precise ser modificado ou poluído.
public interface PagamentoInternacional : Estornavel {
    public fun processarTransacaoInternacional(moedaEstrangeira : String )
}
